File: library/src/test/java/me/proxer/library/enums/LicenseTest.java
package me.proxer.library.enums;

import me.proxer.library.ProxerTest;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

class LicenseTest extends ProxerTest {

    @Test
    void testDefault() throws IOException {
        final License license = api.moshi().adapter(License.class).fromJson("\"2\"");

        assertThat(license).isSameAs(License.LICENSED);
    }

    @Test
    void testFallback() throws IOException {
        final License license = api.moshi().adapter(License.class).fromJson("\"xyz\"");

        assertThat(license).isSameAs(License.UNKNOWN);
    }
}

File: library/src/test/kotlin/me/proxer/library/enums/LicenseTest.kt
package me.proxer.library.enums

import com.serjltt.moshi.adapters.FallbackEnum
import com.squareup.moshi.Moshi
import org.amshove.kluent.shouldBe
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class LicenseTest {

    private val adapter = Moshi.Builder()
        .add(FallbackEnum.ADAPTER_FACTORY)
        .build()
        .adapter(License::class.java)

    @Test
    fun testDefault() {
        adapter.fromJson("\"2\"") shouldBe License.LICENSED
    }

    @Test
    fun testFallback() {
        adapter.fromJson("\"xyz\"") shouldBe License.UNKNOWN
    }
}
